There are a couple compile-time warnings on the Windows build:
```
C:\Code\Python\Python\gc.c(1240,30): warning C4244: 'function': conversion from 'uintptr_t' to 'int', possibl
e loss of data [C:\Code\Python\PCbuild\pythoncore.vcxproj]
```
My guess is that `gc_set_old_space()` should be declared as having a fatter `size` argument than `int`.
```
C:\Code\Python\Python\gc.c(1700,54): warning C4244: 'function': conversion from 'Py_ssize_t' to 'int', possib
le loss of data [C:\Code\Python\PCbuild\pythoncore.vcxproj]
```
And here I guess `_PyGC_GetObjects()`'s `generation` argument should be `int` instead of `Py_ssize_t`.

--- a/Include/internal/pycore_gc.h
+++ b/Include/internal/pycore_gc.h
@@ -322,7 +322,7 @@ extern void _PyGC_Unfreeze(PyInterpreterState *interp);
 /* Number of frozen objects */
 extern Py_ssize_t _PyGC_GetFreezeCount(PyInterpreterState *interp);
 
-extern PyObject *_PyGC_GetObjects(PyInterpreterState *interp, Py_ssize_t generation);
+extern PyObject *_PyGC_GetObjects(PyInterpreterState *interp, int generation);
 extern PyObject *_PyGC_GetReferrers(PyInterpreterState *interp, PyObject *objs);
 
 // Functions to clear types free lists

--- a/Lib/test/test_gc.py
+++ b/Lib/test/test_gc.py
@@ -1094,7 +1094,7 @@ class IncremetalGCTests(unittest.TestCase):
         head = make_ll(1000)
         count = 1000
 
-        # There will some objects we aren't counting,
+        # There will be some objects we aren't counting,
         # e.g. the gc stats dicts. This test checks
         # that the counts don't grow, so we try to
         # correct for the uncounted objects

--- a/Lib/test/test_gc.py
+++ b/Lib/test/test_gc.py
@@ -1059,9 +1059,10 @@ class GCTests(unittest.TestCase):
         gc.enable()
 
 
-class IncremetalGCTests(unittest.TestCase):
+class IncrementalGCTests(unittest.TestCase):
 
     def setUp(self):
+        # Reenable GC as it is disabled module-wide
         gc.enable()
 
     def tearDown(self):

--- a/Modules/gcmodule.c
+++ b/Modules/gcmodule.c
@@ -326,7 +326,7 @@ gc_get_objects_impl(PyObject *module, Py_ssize_t generation)
     }
 
     PyInterpreterState *interp = _PyInterpreterState_GET();
-    return _PyGC_GetObjects(interp, generation);
+    return _PyGC_GetObjects(interp, (int)generation);
 }
 
 /*[clinic input]

--- a/Python/gc.c
+++ b/Python/gc.c
@@ -1233,7 +1233,7 @@ gc_collect_region(PyThreadState *tstate,
                   struct gc_collection_stats *stats);
 
 static inline Py_ssize_t
-gc_list_set_space(PyGC_Head *list, uintptr_t space)
+gc_list_set_space(PyGC_Head *list, int space)
 {
     Py_ssize_t size = 0;
     PyGC_Head *gc;
@@ -1673,7 +1673,7 @@ _PyGC_GetReferrers(PyInterpreterState *interp, PyObject *objs)
 }
 
 PyObject *
-_PyGC_GetObjects(PyInterpreterState *interp, Py_ssize_t generation)
+_PyGC_GetObjects(PyInterpreterState *interp, int generation)
 {
     assert(generation >= -1 && generation < NUM_GENERATIONS);
     GCState *gcstate = &interp->gc;

--- a/Python/gc.c
+++ b/Python/gc.c
@@ -1385,7 +1385,10 @@ expand_region_transitively_reachable(PyGC_Head *container, PyGC_Head *gc, GCStat
 static void
 completed_cycle(GCState *gcstate)
 {
-    assert(gc_list_is_empty(&gcstate->old[gcstate->visited_space^1].head));
+#ifdef Py_DEBUG
+    PyGC_Head *not_visited = &gcstate->old[gcstate->visited_space^1].head;
+    assert(gc_list_is_empty(not_visited));
+#endif
     gcstate->visited_space = flip_old_space(gcstate->visited_space);
     gcstate->work_to_do = 0;
 }

--- a/Python/gc_free_threading.c
+++ b/Python/gc_free_threading.c
@@ -1305,7 +1305,7 @@ visit_get_objects(const mi_heap_t *heap, const mi_heap_area_t *area,
 }
 
 PyObject *
-_PyGC_GetObjects(PyInterpreterState *interp, Py_ssize_t generation)
+_PyGC_GetObjects(PyInterpreterState *interp, int generation)
 {
     PyObject *result = PyList_New(0);
     if (!result) {

